plugins/inputs/unbound/README.md

```diff
@@ -0,0 +1,186 @@
+# Unbound Input Plugin
+
+This plugin gathers stats from [Unbound - a validating, recursive, and caching DNS resolver](https://www.unbound.net/)
+
+### Configuration:
+
+```toml
+ # A plugin to collect stats from Unbound - a validating, recursive, and caching DNS resolver
+ [[inputs.unbound]]
+   ## If running as a restricted user you can prepend sudo for additional access:
+   #use_sudo = false
+
+   ## The default location of the unbound-control binary can be overridden with:
+   binary = "/usr/sbin/unbound-control"
+
+   ## By default, telegraf gathers stats for 3 metric points.
+   ## Setting stats will override the defaults shown below.
+   ## stats may also be set to ["all"], which will collect all stats
+   stats = ["total.*", "num.*","time.up", "mem.*"]
+```
+
+### Measurements & Fields:
+
+This is the full list of stats provided by unbound. Stats will be grouped by their prefix (eg thread0,
+total, etc). In the output, the prefix will be used as a tag, and removed from field names. See
+https://www.unbound.net/documentation/unbound-control.html for details.
+
+- unbound
+    thread0.num.queries
+    thread0.num.cachehits
+    thread0.num.cachemiss
+    thread0.num.prefetch
+    thread0.num.recursivereplies
+    thread0.requestlist.avg
+    thread0.requestlist.max
+    thread0.requestlist.overwritten
+    thread0.requestlist.exceeded
+    thread0.requestlist.current.all
+    thread0.requestlist.current.user
+    thread0.recursion.time.avg
+    thread0.recursion.time.median
+    total.num.queries
+    total.num.cachehits
+    total.num.cachemiss
+    total.num.prefetch
+    total.num.recursivereplies
+    total.requestlist.avg
+    total.requestlist.max
+    total.requestlist.overwritten
+    total.requestlist.exceeded
+    total.requestlist.current.all
+    total.requestlist.current.user
+    total.recursion.time.avg
+    total.recursion.time.median
+    time.now
+    time.up
+    time.elapsed
+    mem.total.sbrk
+    mem.cache.rrset
+    mem.cache.message
+    mem.mod.iterator
+    mem.mod.validator
+    histogram.000000.000000.to.000000.000001
```

Comment:
How are these histogram metrics encoded?  Is the field named `000000.000000.to.000000.000001`?  Maybe we can do special processing of these, normally we use the format produced by the histogram aggregator.

Reply:
Yes actually the field name will be 000000.000000.to.000000.000001 which is not really usefful. Anyway I don't think that this kind of information is relevant for telegraf. It is easier to construct the histogram from telegraf collected data rather than using these fields. I think I will skip them from collected metrics.